db.js

```diff
@@ -312,20 +313,63 @@ exports.getUsers = () => {
   return db.get('users').value()
 }
 
-exports.userHasHighscore = userID => {
-  const highscores = exports.getTopStreaks()
+exports.userHasHighscore = (guildID, userID) => {
+  const highscores = exports.getTopStreaks(guildID)
   if(highscores.length === 0) return false
 
   return Boolean(highscores.find(highscore => {
     return highscore.userID === userID
   }))
 }
 
-exports.getFirstStreakDate = () => {
-  const streaks = db.get('streaks').sortBy('date').value()
+exports.getFirstStreakDate = (guildID) => {
+  const streaks = db.get('streaks').filter(streak => streak.guildID === guildID).sortBy('date').value()
   return format(streaks[0].date, 'do MMM YYYY')
 }
 
 exports.getStreaks = () => {
   return db.get('streaks').value()
+}
+
+/**
+ * Add a role type definition to the database.
+ * @param roleid the id of the role to be added
+ * @param guildid id of the guild in which the role is to be defined
+ * @param type which type of role is to be added
+ */
+exports.addRole = (roleid, guildid, type) => {
```

Comment:
This is gonna sound really pendantic but can we do roleID, guildID, just for consistency